rust/src/dataset.rs

```diff
@@ -508,6 +529,14 @@ impl Dataset {
     }
 }
 
+#[inline]
+fn simd_stride() -> i32 {
```

Comment:
Put into a separate module? and call it `simd_alignment`, which is consistent with the rest of Rust toolchain. https://doc.rust-lang.org/reference/type-layout.html#size-and-alignment

Reply:
put it in distance for now since that's what needs the check